### ISSUE TITLE
Feature: Recursive #transform_values

### ISSUE THREAD

**Issue by Epigene** [zverok/hm#2]
Hi, this gem shows a lot of promise!

I find it would be useful to access values that are not hashes or arrays themselves and apply changes.

For example, I have a hash with various namespaces and generally number values (when they are not nested hashes)

```rb
hash = {
  a: 1.25,
  b: {
    c: 2.13
  }
}
```

It would be great to do something like

```rb
proc = Proc.new{ |v|
  if v.is_a?(Hash) || v.is_a?(Array)
    v
  else
    v.to_s
  end
}

Hm.new(hash).recursively_transform_values(:*, &proc).to_h
#=> {
  a: "1.25",
  b: {c: "2.13"}
}
```

**Comment by zverok** [comment]
Hi, thanks for your suggestion. 
Is it a theoretical case, or you have some real-life situation to require that? 
I thought about this (just as a behavior for argument-less `transform_values`, like `transform_keys` do), but eventually decided against, mostly because it will always require complicated procs ("if it is nested or final value?") like in your example.

**Comment by Epigene** [comment]
It's a live need, I'm working on a statistics panel (a giant table of number data) and it's a hassle to wrap every single value in stringification (with some custom bells and whistles) logic.

I think recursive functionality would come handy, at least with tasks I tend to do.

**Comment by zverok** [comment]
@Epigene Well, after thinking some more, I've released ver 0.0.3, which supports argument-less `transform_values`. In this case, only **terminal** (non-diggable) values will be yield and transformed, so your initial example will be simplified down to:

```ruby
Hm.new(hash).transform_values(&:to_s).to_h
# => {:a=>"1.25", :b=>{:c=>"2.13"}} 
```

Please give it a try and let me know what you think.

**Comment by Epigene** [comment]
Works like a charm! Thanks for this.
Also, thanks for `.transform_values(&SomeClass.method(:some_method))`, I was only familiar with `.map(&:to_s)`
